x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/SourceDestValidator.java

```diff
@@ -26,7 +27,14 @@
  */
 public final class SourceDestValidator {
 
-    private SourceDestValidator() {}
+    interface SourceDestValidation {
+        boolean isDeferrable();
+        void check(DataFrameTransformConfig config, ClusterState clusterState, IndexNameExpressionResolver indexNameExpressionResolver);
```

Comment:
nit: I still find it confusing that I have to call `check` and not `validate` on a "validator". (Vice versa it could also be a `SourceDestChecker` if you prefer to keep `check`)